app/src/main/java/com/example/jwplayerdemo/jwsettings/JWViewModel.java

```diff
@@ -56,76 +74,120 @@
-//                .skinConfig()
-//                .stretching()
-//                .useTextureView()
-                .build();
+        Log.i("HYUNJOO", "JWViewModel - getConfig() ");
+
+        List<PlaylistItem> playlistItemList = new ArrayList<PlaylistItem>(){{
+            add(playlistItem);
+        }};
+
+        config.setPlaylist(playlistItemList);
+
+        return config;
     }
 
     void setFile(String file) {
         JWLogger.log("JWViewModel - Add stream tag: " + file);
-
         if (file.startsWith("{")) {
             JWLogger.log("JWViewModel - Add JSON tag: " + file);
-            this.json = file;
-        } else {
-            this.file = file;
-        }
+            config.setPlaylist(PlaylistItem.listFromJson(file));
+        } else config.setFile(file);
     }
 
     void setMediaId(String mediaid) {
-        JWLogger.log("JWViewModel - Add Media Id: " + mediaid);
     }
 
     void setPlaylistId(String playlistid) {
-        JWLogger.log("JWViewModel - Add Playlist Id: " + playlistid);
     }
 
     void setTitle(String title) {
         JWLogger.log("JWViewModel - Add Title: " + title);
+        playlistItem.setTitle(title);
+        config.setDisplayTitle(checkedTitle);
+    }
+
+    void setDescription(String description) {
+        JWLogger.log("JWViewModel - Add Description: " + description);
+        playlistItem.setDescription(description);
+        config.setDisplayDescription(checkedDescription);
     }
 
     void setImage(String image) {
         JWLogger.log("JWViewModel - Add Image: " + image);
+        if (checkedImage) playlistItem.setImage(image);
     }
 
-    void setDescription(String description) {
-        JWLogger.log("JWViewModel - Add Description: " + description);
+    void showTitle(boolean isChecked) {
+        checkedTitle = isChecked;
+        if(!checkedTitle) playlistItem.setTitle("");
     }
 
-    @Override
-    protected void onCleared() {
-        super.onCleared();
+    void showDescription(boolean isChecked) {
+        checkedDescription = isChecked;
+        if(!checkedDescription) playlistItem.setDescription("");
+    }
+
+    void showImage(boolean isChecked) {
+        checkedImage = isChecked;
+        if(!checkedImage) config.setImage("");
+    }
+
+    boolean getAutoStart() {
+        JWLogger.log("Autostart: " + autostart);
+        return autostart;
+    }
+
+    void setAutoStart(boolean isChecked) {
+        autostart = isChecked;
+        config.setAutostart(isChecked);
+    }
+
+    boolean getRepeat() {
+        JWLogger.log("Repeat mode " + repeat);
+        return repeat;
+    }
+
+    void setRepeat(boolean isChecked) {
+        repeat = isChecked;
+        config.setRepeat(isChecked);
+    }
+
+    boolean getMute() {
+        JWLogger.log("Mute " + mute);
+        return mute;
+    }
+
+    void setMute(boolean isChecked) {
+        mute = isChecked;
+        config.setMute(isChecked);
+    }
+
+
+    boolean mediaIdChecked() {
```

Comment:
Java typically uses the naming convention `isSomething()` for functions that return a boolean that describes state (e.g. `isMediaIdChecked()`)